app/models/chargeback.rb

```diff
@@ -64,6 +68,22 @@ def self.build_results_for_report_chargeback(options)
     [data.map { |r| new(r.last) }]
   end
 
+  def self._get_keys_and_extra_fields(perf, ts_key)
```

Comment:
I'm wonder if this can be cleaned up a bit. Seems weird to have the method named with a leading underscore that will then call the regular method.

May be better to have 2 separate methods `get_keys_and_extra_fields` and `get_keys_and_extra_fields`
Maybe change the caller back to call `get_keys_and_extra_fields` and call one or the other depending on the value of `@options[:groupby_tag]`